test/config/DocumentationControllerSpec.scala

```diff
@@ -16,23 +16,40 @@
 
 package config
 
-import play.api.test.Helpers._
 import com.typesafe.config.ConfigFactory
 import controllers.Assets
 import definition.ApiDefinitionFactory
-import mocks.MockAppConfig
-import play.api.{Application, Configuration}
-import utils.BaseSpec
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.{reset, when}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatestplus.mockito.MockitoSugar.mock
 import play.api.http.{DefaultHttpErrorHandler, HttpErrorConfig}
 import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.libs.json.Json
 import play.api.mvc.{ControllerComponents, Result}
+import play.api.test.Helpers._
+import play.api.{Application, Configuration}
+import utils.BaseSpec
 
 import scala.concurrent.Future
 
-class DocumentationControllerSpec extends BaseSpec {
+class DocumentationControllerSpec extends BaseSpec with BeforeAndAfterEach {
+
+  private val mockAppConfig: AppConfig = mock[AppConfig]
 
-  private val selfAssessmentApiDefinition: ApiDefinitionFactory = new ApiDefinitionFactory(TestHelper.mockConfig)
+  override def beforeEach(): Unit = {
+    reset(mockAppConfig)
+
+    when(mockAppConfig.apiGatewayContext).thenReturn("gateway")
+    when(mockAppConfig.featureSwitch).thenReturn(Some(Configuration(ConfigFactory.parseString("""
+                                                                                         |version-1.enabled = true
+                                                                                         |version-2.enabled = true
+      """.stripMargin))))
+    when(mockAppConfig.apiStatus(ArgumentMatchers.eq("1.0"))).thenReturn("ALPHA")
+    when(mockAppConfig.endpointsEnabled).thenReturn(true)
+  }
```

Comment:
I feel you don't need beforeEach and need to reset as the values here are not updated. All mocking can be the below without the other stuff.
```

    when(mockAppConfig.apiGatewayContext).thenReturn("gateway")
    when(mockAppConfig.featureSwitch).thenReturn(Some(Configuration(ConfigFactory.parseString("""
        |version-1.enabled = true
        |version-2.enabled = true
      """.stripMargin))))
    when(mockAppConfig.apiStatus(ArgumentMatchers.eq("1.0"))).thenReturn("ALPHA")
    when(mockAppConfig.endpointsEnabled).thenReturn(true)
```
or can put it in a trait and use new keyword

```

  private trait TestSetup {
    when(mockAppConfig.apiGatewayContext).thenReturn("gateway")
    when(mockAppConfig.featureSwitch).thenReturn(Some(Configuration(ConfigFactory.parseString("""
        |version-1.enabled = true
        |version-2.enabled = true
      """.stripMargin))))
    when(mockAppConfig.apiStatus(ArgumentMatchers.eq("1.0"))).thenReturn("ALPHA")
    when(mockAppConfig.endpointsEnabled).thenReturn(true)
  }
```